packages/core/src/app/ovm/deciders/db-decider.ts

```diff
@@ -0,0 +1,70 @@
+import { Md5 } from 'ts-md5/dist/md5'
+
+import { Decider, Decision } from '../../../types/ovm/decider.interface'
+import { Bucket, DB } from '../../../types/db'
+
+export abstract class DbDecider implements Decider {
+  private readonly decisionBucket: Bucket
+
+  protected constructor(db: DB) {
+    this.decisionBucket = db.bucket(Buffer.from(this.getUniqueId()))
+  }
+
+  public async checkDecision(input: any): Promise<Decision> {
+    const hash: Buffer = DbDecider.getCacheKey(input)
+    const decisionBuffer: Buffer = await this.decisionBucket.get(hash)
+
+    if (decisionBuffer === null) {
+      return undefined
+    }
+
+    return this.deserializeDecision(decisionBuffer)
+  }
+
+  /**
+   * Stores the provided decision for the provided input.
+   *
+   * @param input the input that resulted in the provided decision
+   * @param serializedDecision the buffer representing the Decision to be stored
+   */
+  protected async storeDecision(
+    input: any,
+    serializedDecision: Buffer
+  ): Promise<void> {
+    const key: Buffer = DbDecider.getCacheKey(input)
+
+    await this.decisionBucket.put(key, serializedDecision)
+  }
+
+  /**
+   * Gets the unique key for the provided input to use as a cache key for its Decisions
+   *
+   * @param input the input for which a key will be computed
+   * @returns the computed cache key
+   */
+  private static getCacheKey(input: any): Buffer {
+    return Buffer.from(Md5.hashStr(JSON.stringify(input)) as string)
```

Comment:
Yeah, can certainly do that. Could also change it here down the line. Really just looking for something reasonably fast that produces unique finite-sized output. I'll pull it out into crypto.ts and/or have it take in a hash function as a dependency.

Reply:
For transparency, I kept it as non-configurable because there's no reason an implementation should need to configure it (or even know how keys are generated).